src/cache/cache_las.c

```diff
@@ -1078,12 +1077,16 @@ __wt_las_sweep(WT_SESSION_IMPL *session)
 			cnt = 0;
 
 		/*
-		 * If we have processed enough entries and we are between
-		 * blocks, give up.
+		 * If we have processed enough entries and there is a reader
+		 * wanting to get the lock or we are between blocks, give up.
 		 */
-		if (cnt > 0)
+		++visit_cnt;
+		if (cnt > 0) {
 			--cnt;
-		else if (saved_key->size == 0)
+			if (visit_cnt > WT_LAS_SWEEP_ENTRIES &&
+			    cache->las_reader)
```

Comment:
Even with this condition (which i like), it's still potentially going to do twice as much work as before between relinquishing the lock. Do you think it makes sense to add another layer that releases the lock every ~1000 ops if there are readers waiting, but doesn't go back to sleep until it's done `cnt` things?

Reply:
First, I'm happy to set the entries back to 10000 if you want. I think it was fairly arbitrarily chosen and 20000 is more but doesn't feel like too much either. I'm hesitant to keep it in there until `cnt` things is complete if it is getting interrupted or slowed down because sweep does need to get back to the caller and do the other sweep things. And another layer is just more complex.

Reply:
Also, I don't feel like this is quite right yet. The current algorithm only breaks on a keyspace boundary allowing the count to stay zero while we process up to the boundary. I think the new code should do the same. I think other parts of the loop expect stopping on the boundary.

Reply:
I think I fixed the boundary issue.